pkg/util/provider/machinecontroller/controller.go

```diff
@@ -169,18 +189,25 @@ func NewController(
 	})
 
 	// MachineSafety Controller Informers
-
 	// We follow the kubernetes way of reconciling the safety controller
 	// done by adding empty key objects. We initialize it, to trigger
 	// running of different safety loop on MCM startup.
 	controller.machineSafetyOrphanVMsQueue.Add("")
 	controller.machineSafetyAPIServerQueue.Add("")
-
 	machineInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		// deleteMachineToSafety makes sure that orphan VM handler is invoked
 		DeleteFunc: controller.deleteMachineToSafety,
 	})
 
+	// Drain Controller Informers
+	if k8sutils.IsResourceSupported(targetCoreClient, volumeAttachmentGroupName, volumeAttachmentResourceName, volumeAttachmentResourceKind) {
+		controller.volumeAttachmentHandler = drain.NewVolumeAttachmentHandler()
+		volumeAttachmentInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc:    controller.volumeAttachmentHandler.AddVolumeAttachment,
+			UpdateFunc: controller.volumeAttachmentHandler.UpdateVolumeAttachment,
```

Comment:
Wouldn't complete the functionality if we use the `DeleteFunc` for the `waitForDetach` as well? Or do you want to keep it for later cleanup?